TEST/TEST_knapsack.cpp

```diff
@@ -32,19 +34,34 @@ void TEST_KNAPSACK_EXACT() {
   const float capacity = 8.8f;
 
   AAL_knapsackInstance instance{itemSizes, itemValues, numberOfItems, capacity};
-  AAL_knapsackResult result = AAL_knapsack_exact(instance);
 
   unsigned int* subset = (unsigned int*) malloc(3 * sizeof(unsigned int));
   subset[0]            = 0;
   subset[1]            = 5;
   subset[2]            = 7;
 
-  print_check(
-    std::abs(result.objectiveValue - 13.1f) < 0.00001f && compareAryUInt(result.items, result.numberOfItems, subset, 3),
-    "AAL_KNAPSACK_EXACT");
+  if (testset.count("KSE") != 0) {
```

Comment:
The knapsack test does not run if for example `FULL` or `AAL` are given. I think the best way is to set the flags earlier in the function and then execute based on if the respective flag is set.

Reply:
The highlighted code by Github is a bit misleading, I simply mean that the if does not check for all cases.